include/mxnet/io.h

```diff
@@ -52,7 +52,7 @@ class IIterator : public dmlc::DataIter<DType> {
   /*! \brief move to next item */
   virtual bool Next(void) = 0;
   /*! \brief get current data */
-  virtual const DType &Value(void) const = 0;
```

Comment:
I think both styles are fine: https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions

No need to change this